### PR TITLE
Sage/building a smart contract lottery application on celo with python

### PR DESCRIPTION
**Pull request by Southpaw0**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+---
+
+![header](../../src/data-tutorials/showcase/intermediate/building-a-smart-contract-lottery-application-on-celo-with-python.png)
+
+## Introduction
+
+Lottery applications are one of the major applications of blockchain technology since the early days of Ethereum. Smart contracts enable the operation of secure and transparent lotteries that cannot be manipulated by centralized parties. Furthermore, they can automate many lottery-related processes, such as ticket sales, winner selection, and prize distribution.
+In this tutorial, we will learn how to to build a lottery smart contract and deploy it on the Celo blockchain with Python using brownie framework.
+
+## Prerequisites[](https://docs.celo.org/community/celo-sage/tutorial-structure#prerequisites)
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:22]:
```suggestion
## Prerequisites
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+To understand this tutorial, you must be familiar with:
+
+- Building Smart contracts
+- The Python Programming Language
+
+## Requirements[](https://docs.celo.org/community/celo-sage/tutorial-structure#requirements)
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:29]:
```suggestion
## Requirements
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+After installing dependencies, we need to initialize our project as a brownie project.
+
+```bash
+brownie init
+```
+
+This command generates some folders which look like this:
+
+<img width="271" alt="Screenshot 2023-03-16 at 2 13 03 PM" src="https://user-images.githubusercontent.com/104994589/225643776-acffdc8e-371d-4668-8799-601a51e55c8e.png">
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:76]:
```suggestion
![Screenshot](https://user-images.githubusercontent.com/104994589/225643776-acffdc8e-371d-4668-8799-601a51e55c8e.png)
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+brownie networks add Celo celo-alfajores host=https://alfajores-forno.celo-testnet.org chainid=44787 explorer=https://alfajores-blockscout.celo-testnet.org
+```
+
+You can see the list of networks that have been added to our brownie project:
+
+```bash
+brownie networks list
+```
+
+<img width="1127" alt="Screenshot 2023-03-16 at 2 15 50 PM" src="https://user-images.githubusercontent.com/104994589/225643608-6a3645a6-f0b7-4588-b975-219c8e950fb3.png">
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:122]:
```suggestion
![Screenshot](https://user-images.githubusercontent.com/104994589/225643608-6a3645a6-f0b7-4588-b975-219c8e950fb3.png)
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+        require(msg.sender == manager, "Only the manager can call this function.");
+        _;
+    }
+}
+```
+
+Let’s go through the code step by step:
+
+```solidity
+		address public manager;
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:185]:
```suggestion
	address public manager;
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+The deploy_lottery function gets the account we would use to deploy the contract.
+
+Deploy the contract:
+
+```bash
+
+```
+
+<img width="1127" alt="Screenshot 2023-03-16 at 2 57 57 PM" src="https://user-images.githubusercontent.com/104994589/225643259-30040976-23a9-41b3-a03e-ddf008850229.png">
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:291]:
```suggestion
![Screenshot](https://user-images.githubusercontent.com/104994589/225643259-30040976-23a9-41b3-a03e-ddf008850229.png)
```

File: src/building-a-smart-contract-lottery-application-on-celo-with-python.png
Comment (developerkunal) [inline src/building-a-smart-contract-lottery-application-on-celo-with-python.png:1]: put the image in intermediate folder

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+
+After installing dependencies, we need to initialize our project as a brownie project.
+
+```bash
+brownie init
+```
+
+This command generates some folders which look like this:
+
++![Screenshot](https://user-images.githubusercontent.com/104994589/225643776-acffdc8e-371d-4668-8799-601a51e55c8e.png)
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:76]:
```suggestion
![Screenshot](https://user-images.githubusercontent.com/104994589/225643776-acffdc8e-371d-4668-8799-601a51e55c8e.png)
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+
+### Deploying the Contract
+
+Next, we need to compile and deploy the contract on the Celo Testnet, Run the following command to compile the contract.
+
+```bash
+brownie compile
+```
+
+<img width="1127" alt="Screenshot 2023-03-16 at 2 37 58 PM" src="https://user-images.githubusercontent.com/104994589/225643393-810bd489-c110-47cd-95b3-9028862f49d6.png">
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:254]:
```suggestion
![Screenshot](https://user-images.githubusercontent.com/104994589/225643393-810bd489-c110-47cd-95b3-9028862f49d6.png)
```

File: blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+brownie networks add Celo celo-alfajores host=https://alfajores-forno.celo-testnet.org chainid=44787 explorer=https://alfajores-blockscout.celo-testnet.org
+```
+
+You can see the list of networks that have been added to our brownie project:
+
+```bash
+brownie networks list
+```
+
++![Screenshot](https://user-images.githubusercontent.com/104994589/225643608-6a3645a6-f0b7-4588-b975-219c8e950fb3.png)
```
Comment (developerkunal) [inline blog/sage-building-a-smart-contract-lottery-application-on-celo-with-python/sage-building-a-smart-contract-lottery-application-on-celo-with-python.md:122]:
```suggestion
![Screenshot](https://user-images.githubusercontent.com/104994589/225643608-6a3645a6-f0b7-4588-b975-219c8e950fb3.png)
```

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**